lib/pengine/utils.c

```diff
@@ -688,35 +688,35 @@ pe__update_recheck_time(time_t recheck, pcmk_scheduler_t *scheduler,
  *
  * \param[in]     xml_obj       XML element containing blocks of nvpair elements
  * \param[in]     set_name      If not NULL, only use blocks of this element
- * \param[in]     rule_data     Matching parameters to use when unpacking
- *                              (node_hash member must be NULL if \p set_name is
- *                              PCMK_XE_META_ATTRIBUTES)
+ * \param[in]     rule_input    Values used to evaluate rule criteria
+ *                              (node_attrs member must be NULL if \p set_name
+ *                              is PCMK_XE_META_ATTRIBUTES)
  * \param[out]    hash          Where to store extracted name/value pairs
  * \param[in]     always_first  If not NULL, process block with this ID first
  * \param[in,out] scheduler     Scheduler data containing \p xml_obj
  */
 void
 pe__unpack_dataset_nvpairs(const xmlNode *xml_obj, const char *set_name,
-                           const pe_rule_eval_data_t *rule_data,
+                           const pcmk_rule_input_t *rule_input,
                            GHashTable *hash, const char *always_first,
                            pcmk_scheduler_t *scheduler)
 {
     crm_time_t *next_change = NULL;
 
-    CRM_CHECK((set_name != NULL) && (rule_data != NULL) && (hash != NULL)
+    CRM_CHECK((set_name != NULL) && (rule_input != NULL) && (hash != NULL)
               && (scheduler != NULL), return);
 
     // Node attribute expressions are not allowed for meta-attributes
-    CRM_CHECK((rule_data->node_hash == NULL)
+    CRM_CHECK((rule_input->node_attrs == NULL)
```

Comment:
Just an arbitrary location I'm stumbling over probably ...
Shouldn't we for such cases rather go for a logical expression that resembles the natural language above?
!((rule_input->node_attrs != NULL) && (strcmp(set_name, PCMK_XE_META_ATTRIBUTES) == 0))
Even nicer of course the other way round but I guess the optimization wouldn't be allowed then to do the simple check first.

Reply:
meh, not worth it